magenta-lib/src/main/scala/magenta/deployment_type/S3.scala

```diff
@@ -106,18 +107,18 @@ object S3 extends DeploymentType with S3AclParams {
         assert(data.isDefined, s"Cannot find resource value for ${bucketResource(pkg)} (${pkg.apps.head} in ${parameters.stage.name})")
         data.get.value
       }
+      val prefix:String = S3UploadV2.prefixGenerator(
+        stack = if (prefixStack(pkg)) Some(stack) else None,
+        stage = if (prefixStage(pkg)) Some(parameters.stage) else None,
+        packageName = if (prefixPackage(pkg)) Some(pkg.name) else None
+      )
```

Comment:
After more thought I've decided not to bother DRYing this out. It is minimal code duplication.